### PR TITLE
Add ttyd to conda-forge

### PR DESCRIPTION
**Pull request by sodre**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by sodre** [comment]
ttyd depends on xxd (https://sourceforge.net/p/xxd/code/ref/master/). This is usually included as part of vim-commons.

File: recipes/ttyd/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - make 
+    - make install
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake 
+    - pkg-config
+    - vim
```
Comment (scopatz) [inline recipes/ttyd/meta.yaml]: Should vim be a host dep?

File: recipes/ttyd/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+build:
+  number: 0
+  skip: true  # [not linux]
+  script:
+    - mkdir build
+    - cd build
+    - cmake -DCMAKE_INSTALL_PREFIX={{PREFIX}} -DCMAKE_PREFIX_PATH={{PREFIX}} ..
+    - make 
+    - make install
```
Comment (scopatz) [inline recipes/ttyd/meta.yaml]: This should be in a build.sh script

File: recipes/ttyd/meta.yaml (hunk 1 above)
Comment (sodre) [reply recipes/ttyd/meta.yaml]: Not in this case. The project has a dependency on the `xxd` executable during build time only, not a link dependency.

Comment (sodre) [reply recipes/ttyd/meta.yaml]: The `ttyd` [CMakeLists.txt][link] file uses `xxd` only during build time. This is lifted from their code.

```cmake
find_program(CMAKE_XXD NAMES xxd)
add_custom_command(OUTPUT html.h
        COMMAND ${CMAKE_XXD} -i index.html html.h
        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/src
        COMMENT "Generating html.h from index.html")
list(APPEND SOURCE_FILES html.h)
```
[link]: https://github.com/tsl0922/ttyd/blob/01a2ce07257bc5f6be02e8b870f6eab0d03c8d23/CMakeLists.txt#L62-L67

Comment (scopatz) [reply recipes/ttyd/meta.yaml]: Doesn't that make it a host dep?

Comment (sodre) [reply recipes/ttyd/meta.yaml]: I don't think so, according to [this post][post] by isuruf, you should move something to host if it is used as a library, which is not the case here.

The way I understood it from the [documentation][doc], we put a tool in `build` if we need it executed during build time, but it won't be needed by the end-package during runtime, e.g. cmake.

[doc]: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#build
[post]: https://github.com/conda-forge/staged-recipes/pull/9136#discussion_r315531004

Comment (scopatz) [reply recipes/ttyd/meta.yaml]: Ahh I see it doesn't contain a header file but is only used to generate the header. Yeah this is fine.

**Pull request merged**